File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        arr = [(nums[i], i) for i in range(n)]
        arr.sort()

        i, j = 0, n - 1
        while i < j:
            num1, index1 = arr[i]
            num2, index2 = arr[j]
            total = num1 + num2
            if total == target:
                return [index1, index2]
            elif total < target:
                i += 1
            else:
                j -= 1

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> cache;
        for (int i = 0; i < nums.size(); i++) {
            int comp = target - nums[i];
            if (cache.contains(comp))
                return {cache[comp], i};
            cache[nums[i]] = i;
        }
        return {};
    }
};